crates/symbolicator/src/services/download/mod.rs

```diff
@@ -35,20 +36,27 @@ const USER_AGENT: &str = concat!("symbolicator/", env!("CARGO_PKG_VERSION"));
 /// Errors happening while downloading from sources.
 #[derive(Debug, Error)]
 pub enum DownloadError {
-    #[error("failed to download")]
+    #[error("failed to download (IO)")]
     Io(#[source] std::io::Error),
-    #[error("failed to download")]
+    #[error("failed to download (streaming file)")]
     Reqwest(#[source] reqwest::Error),
     #[error("bad file destination")]
     BadDestination(#[source] std::io::Error),
     #[error("failed writing the downloaded file")]
     Write(#[source] std::io::Error),
     #[error("download was cancelled")]
     Canceled,
-    #[error("failed to fetch data from GCS")]
+    #[error("failed to fetch data from GCS: {0}")]
```

Comment:
Per [this guideline](https://blog.rust-lang.org/inside-rust/2021/07/01/What-the-error-handling-project-group-is-working-towards.html#guidelines-for-implementing-displayfmt-and-errorsource) you should either incorporate the source error's message in this error's `Display` impl or return the source error via `Error::source`, but not both. The same goes for the `Sentry` and `S3` cases below.

Reply:
i've done something a little different here in resolving this: the `Display` impl no longer references the source error, but there now exists a helper `for_cache` on the error enum that constructs a string which does embed the source.

i'm interested in feedback on this: does the new approach make sense, or does it just introduce another unnecessary representation of the error? should the cache store the entire stack of error sources, drawing inspiration from `ObjectError`'s `Display` impl or is it fine storing only the source 1 level down? do error sources only 1 level below provide so little information that this provides no additional useful information?

Reply:
i'm going to assume based off of the silence here that the current approach is okay

Reply:
GitHub won't let me merge until the person who proposed these changes approves the PR 🙃 @loewenheim could I bug you to take a second look at this?